preview-env/create/action.yml

```diff
@@ -68,10 +68,38 @@ runs:
       ref: ${{ inputs.revision }}
   - name: Deploy
     shell: bash
-    run: >-
-      argocd app create ${{ inputs.argocd_arguments }}
+    run: |
+      max_retries=3
+      for attempt in $(seq "$max_retries"); do
+          argocd app create ${{ inputs.argocd_arguments }}
+          if [ $? -eq 0 ]; then
+            echo "Creating Argo CD app with name ${{ inputs.app_name }}"
+            sleep 60
 
-      argocd app sync ${{ inputs.app_name }} --async --force || true
+            # Fetch app information to make sure the app is registered
+            app_info=$(argocd app get ${{ inputs.app_name }} --output json 2>/dev/null)
+            get_status_status=$?
+            if [ $get_status_status -eq 0 ] && [ -n "$app_info" ]; then
+              echo "App Info: $app_info"
```

Comment:
I'd suggest wrapping this `echo` statement in a group so it's collapsed in the workflow log and can be uncollapsed if needed. Makes the output more readable.

```
echo "::group::app info"
echo "App Info: $app_info"
echo "::endgroup::"
```

Reply:
nothing about `--upsert`, was just an observation on how / why the for loop works.
`--upsert` is part of `${{ inputs.argocd_arguments }}` meaning --> `argocd app create ${{ inputs.argocd_arguments }}`.

`--upsert` overrides the existing app, so in case the app exists --> overwritten, if it doesn't exist --> created.
Either way the return code is `0`, which is important for your for loop to work in case the app registration didn't work immediately.

Nothing to change / adjust.

Reply:
ah that makes sense 👍🏻

Reply:
logs look good now https://github.com/camunda/operate/actions/runs/6877333070/job/18848699317

Reply:
sweet :rocket:, looks a lot more clean now. Thank you!